nats/aio/client.py

```diff
@@ -233,20 +234,32 @@ class _SubscriptionMessageIterator:
     def __init__(self, queue):
         self._queue = queue
         self._unsubscribed_future = asyncio.Future()
+        self._iterator_future = asyncio.Future()
 
     def _cancel(self):
-        self._unsubscribed_future.set_result(True)
+        if not self._unsubscribed_future.done():
+            self._unsubscribed_future.set_result(True)
 
     def __aiter__(self):
         return self
 
     async def __anext__(self):
+        if self._iterator_future.done():
+            raise StopAsyncIteration
+
+        if self._unsubscribed_future.done():
```

Comment:
I'm not really sure why two futures are necessary, the wait should be interrupted if the future is completed so this won't block for forever. If we want to exit as soon as possible I think checking the one future should be enough. Not sure what I'm missing?

Reply:
You are right, not needed now so took the extra future out. I was getting a task pending error before but they went away once I added the `elif self._unsubscribed_future.done()` check below:

```python
Task was destroyed but it is pending!
...
  File "/python/nats.py/nats/aio/client.py", line 255, in __anext__
    get_task = asyncio.create_task(self._queue.get())
```